### PR TITLE
PLT-6823 Removes option to set notification sound in Edge

### PR DESCRIPTION
**Pull request by Whiteaj36**
#### Summary
Removes option to set notification sound in Edge. Prompts to use browser settings instead.

#### Ticket Link
[PLT-6823](https://mattermost.atlassian.net/browse/PLT-6823)

#### Checklist
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

File: webapp/utils/utils.jsx [hunk 1]
```diff
@@ -165,6 +165,10 @@ export function ding() {
     }
 }
 
+export function hasNoSoundOptions() {
```
Comment (saturninoabril) [inline webapp/utils/utils.jsx]: Thanks @Whiteaj36 for your contribution! Looks good to me except for minor item. I suggest to name this function as `hasSoundOptions` and subsequently update its logic implementation.

File: webapp/i18n/en.json [hunk 2]
```diff
@@ -2300,16 +2300,16 @@
   "user.settings.notifications.commentsNever": "Do not trigger notifications on messages in reply threads unless I'm mentioned",
   "user.settings.notifications.commentsRoot": "Trigger notifications on messages in threads that I start",
   "user.settings.notifications.desktop": "Send desktop notifications",
-  "user.settings.notifications.desktop.allFirefoxForever": "For all activity, shown indefinitely",
-  "user.settings.notifications.desktop.allFirefoxTimed": "For all activity, shown for {seconds} seconds",
+  "user.settings.notifications.desktop.allSansSoundForever": "For all activity, shown indefinitely",
```
Comment (hmhealey) [inline webapp/i18n/en.json]: Can you change the translation ids to be like `allSoundHiddenForever`? Translators might have difficulty understanding the "sans" part (unless of course they're translating into French 😄)

File: webapp/utils/utils.jsx (hunk 1 above)
Comment (hmhealey) [inline webapp/utils/utils.jsx]: Agreed. It would be nice to use the positive version wherever possible

File: webapp/i18n/en.json (hunk 2 above)
Comment (Whiteaj36) [inline webapp/i18n/en.json]: I had thought that when I wrote it but was having trouble thinking of a good ID. Thanks for the suggestion! 👍

File: webapp/utils/utils.jsx (hunk 1 above)
Comment (Whiteaj36) [inline webapp/utils/utils.jsx]: Can do!

**Comment by grundleborg** [comment]
@Whiteaj36 the build server is reporting that you've got an eslint issue:

```
$ eslint --ext ".jsx" --ignore-pattern node_modules --quiet . 

/home/ubuntu/workspace/mattermost-platform-pr/src/github.com/mattermost/platform/webapp/components/user_settings/desktop_notification_settings.jsx
  47:13  error  Unexpected negated condition  no-negated-condition

✖ 1 problem (1 error, 0 warnings)
```

**Comment by Whiteaj36** [comment]
I see the build issues. Should I solve this by adding an ignore or by restructuring the if blocks?

**Comment by saturninoabril** [comment]
@Whiteaj36 It would be better to restructure the if block and starts with:
```
if ((Utils.hasSoundOptions())) {
```

**Comment by hmhealey** [comment]
Regarding @saturninoabril's comment, the build for this is actually failing ESLint because it expects the if statement to start with `if (Utils.hasSoundOptions())`
```
$ eslint --ext ".jsx" --ignore-pattern node_modules --quiet . 

/home/ubuntu/workspace/mattermost-platform-pr/src/github.com/mattermost/platform/webapp/components/user_settings/desktop_notification_settings.jsx
  47:13  error  Unexpected negated condition  no-negated-condition

✖ 1 problem (1 error, 0 warnings)
```

**Comment by Whiteaj36** [comment]
For sure! Thanks for the direction :) Sorry this took so long.

File: webapp/components/user_settings/desktop_notification_settings.jsx [hunk 3]
```diff
@@ -44,24 +44,7 @@ export default class DesktopNotificationSettings extends React.Component {
...
-                        </label>
-                        <br/>
-                        <FormattedMessage
-                            id='user.settings.notifications.soundConfig'
-                            defaultMessage='Please configure notification sounds in your browser settings'
-                        />
-                    </div>
-                );
-            } else {
+            if ((Utils.hasSoundOptions())) {
```
Comment (saturninoabril) [inline webapp/components/user_settings/desktop_notification_settings.jsx]: Should change to: 
```
if (Utils.hasSoundOptions()) {
```

File: webapp/components/user_settings/desktop_notification_settings.jsx [hunk 4]
```diff
@@ -303,26 +303,26 @@ export default class DesktopNotificationSettings extends React.Component {
     buildMinimizedSetting() {
         let describe = '';
         if (this.props.activity === 'mention') {
-            if (!(Utils.hasSoundOptions())) {
-                if (this.props.duration === '0') {
+            if ((Utils.hasSoundOptions()) && (this.props.sound !== 'false')) {
```
Comment (saturninoabril) [inline webapp/components/user_settings/desktop_notification_settings.jsx]: Should change to:
```
if (Utils.hasSoundOptions() && this.props.sound !== 'false') {
```

File: webapp/components/user_settings/desktop_notification_settings.jsx [hunk 5]
```diff
@@ -303,26 +303,26 @@ export default class DesktopNotificationSettings extends React.Component {
...
+                            id='user.settings.notifications.desktop.mentionsSoundTimed'
+                            defaultMessage='For mentions and direct messages, with sound, shown for {seconds} seconds'
                             values={{
                                 seconds: this.props.duration
                             }}
                         />
                     );
                 }
-            } else if (this.props.sound === 'false') {
+            } else if ((Utils.hasSoundOptions()) && (this.props.sound === 'false')) {
```
Comment (saturninoabril) [inline webapp/components/user_settings/desktop_notification_settings.jsx]: Should change to:
```
} else if (Utils.hasSoundOptions() && this.props.sound === 'false')
```

File: webapp/components/user_settings/desktop_notification_settings.jsx [hunk 6]
```diff
@@ -369,26 +369,26 @@ export default class DesktopNotificationSettings extends React.Component {
                 />
             );
         } else {
-            if (!(Utils.hasSoundOptions())) {  //eslint-disable-line no-lonely-if
-                if (this.props.duration === '0') {
+            if ((Utils.hasSoundOptions()) && (this.props.sound !== 'false')) {  //eslint-disable-line no-lonely-if
```
Comment (saturninoabril) [inline webapp/components/user_settings/desktop_notification_settings.jsx]: Should change to: 
```
if (Utils.hasSoundOptions() && this.props.sound !== 'false') {  
```

File: webapp/components/user_settings/desktop_notification_settings.jsx [hunk 7]
```diff
@@ -369,26 +369,26 @@ export default class DesktopNotificationSettings extends React.Component {
...
+                            id='user.settings.notifications.desktop.allSoundTimed'
+                            defaultMessage='For all activity, with sound, shown for {seconds} seconds'
                             values={{
                                 seconds: this.props.duration
                             }}
                         />
                     );
                 }
-            } else if (this.props.sound === 'false') {
+            } else if ((Utils.hasSoundOptions()) && (this.props.sound === 'false')) {
```
Comment (saturninoabril) [inline webapp/components/user_settings/desktop_notification_settings.jsx]: Should change to:
```
} else if (Utils.hasSoundOptions() && this.props.sound === 'false')
```

Comment (Whiteaj36) [inline webapp/components/user_settings/desktop_notification_settings.jsx]: Thanks! I'm always a bit over-cautious with paranthesis.

**Comment by hmhealey** [comment]
Awesome, thanks for the PR!

**Pull request merged**